protocol/messenger_wallet.go

```diff
@@ -0,0 +1,174 @@
+package protocol
+
+import (
+	"context"
+	"time"
+
+	"github.com/golang/protobuf/proto"
+	"go.uber.org/zap"
+
+	ethcommon "github.com/ethereum/go-ethereum/common"
+	"github.com/status-im/status-go/eth-node/types"
+	"github.com/status-im/status-go/multiaccounts/accounts"
+	"github.com/status-im/status-go/protocol/common"
+	"github.com/status-im/status-go/protocol/protobuf"
+)
+
+var checkBalancesInterval = time.Minute * 10
+
+func (m *Messenger) retrieveWalletBalances() error {
```

Comment:
Why cannot this be in wallet service? I don't think it uses anything from the messenger?
We also have a "Start" function in the wallet (which only emit a tick)

Reply:
I think eventually this will have to check some data from messenger, for example, you might want to be notified when you are able to access a given community, or the opposite, similarly we will want to show notifications to the user etc, so I would think that we might want the loop to be in messenger as it will do all the coordination, what do you think?

Reply:
Regarding the "show notifications" we do have event being sent from status-go to mobile from the wallet. But I guess you mean  notification via a "notification center" ?

It feels weird to me to have this here but if we can only send notification via the messenger then it is the blocker?
Shall we make notification a "global" thing?

Reply:
Messenger should know about notifications, if it doesn't then we would get into issues when you have to sync them between devices, since the flow would be:

Network -> SyncNotification -> Messenger -> ActivityCenter

So messenger is essentially always involved (it's the orchestrator), at least that's the way I see it, which should avoid circular dependencies (you can break down packages further and remove logic from messenger, that makes sense, but the top level object I think should be messenger)